### PR TITLE
Fixing coverity report issues [6100]

### LINKED ISSUE #627: Coverity report reverse null
Should `if(change != NULL)` be checked in the beginning?

https://github.com/eProsima/Fast-RTPS/blob/a109baacad040d549715a965083a86aaa77d6b1b/src/cpp/qos/ParameterList.cpp#L828-L855

### LINKED ISSUE #628: Coverity report resource leak
It seems `GroupDataQosPolicy*p = new GroupDataQosPolicy();` did get delete
https://github.com/eProsima/Fast-RTPS/blob/a109baacad040d549715a965083a86aaa77d6b1b/src/cpp/rtps/builtin/data/WriterProxyData.cpp#L227-L230

### LINKED ISSUE #629: Data leak
`result` did not get released should call `freeaddrinfo()` in the end
https://github.com/eProsima/Fast-RTPS/blob/a109baacad040d549715a965083a86aaa77d6b1b/src/cpp/utils/IPFinder.cpp#L305-L317

same here
https://github.com/eProsima/Fast-RTPS/blob/a109baacad040d549715a965083a86aaa77d6b1b/src/cpp/utils/IPFinder.cpp#L335-L347

### LINKED ISSUE #630: Coverity report potential buffer overflow
1) Event overflow_before_widen: | Potentially overflowing expression "count * original_change_cit->getChange()->getFragmentSize()" with type "unsigned int" (32 bits, unsigned) is evaluated using 32-bit arithmetic, and then used in a context that expects an expression of type "size_t" (64 bits, unsigned).
(2) Event remediation: | To avoid overflow, cast either "count" or "original_change_cit->getChange()->getFragmentSize()" to type "size_t".

https://github.com/eProsima/Fast-RTPS/blob/a109baacad040d549715a965083a86aaa77d6b1b/src/cpp/rtps/reader/FragmentedChangePitStop.cpp#L80-L81

### LINKED ISSUE #631: Coverity report potential infinite loop
Not a big issue, but Loop bound "length" has type "uint32_t" and "i" has type "int16_t", "length" has a bigger upper bound depends on the value it can cause infinite loop
https://github.com/eProsima/Fast-RTPS/blob/a109baacad040d549715a965083a86aaa77d6b1b/include/fastrtps/rtps/messages/CDRMessage.hpp#L89-L92

### PR DESCRIPTION
**Pull request by MiguelCompany**
This should fix #627, fix #628, fix #629, fix #630 and fix #631

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/142/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/131/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/141/)

File: src/cpp/rtps/reader/FragmentedChangePitStop.cpp [hunk 1]
```diff
@@ -68,17 +68,23 @@ CacheChange_t* FragmentedChangePitStop::process(CacheChange_t* incoming_change,
...
             }
             // Last fragment is a special case when copying.
             else
             {
-                memcpy(original_change_cit->getChange()->serializedPayload.data + count * original_change_cit->getChange()->getFragmentSize(),
-                        incoming_change->serializedPayload.data + (count - (fragmentStartingNum - 1)) * incoming_change->getFragmentSize(), original_change_cit->getChange()->serializedPayload.length - (count * original_change_cit->getChange()->getFragmentSize()));
+                memcpy(original_change_cit->getChange()->serializedPayload.data + original_offset,
+                        incoming_change->serializedPayload.data + incoming_offset,
+                        original_change_cit->getChange()->serializedPayload.length - 
+                            (size_t(count) * original_change_cit->getChange()->getFragmentSize()));
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply src/cpp/rtps/reader/FragmentedChangePitStop.cpp]: done

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/156/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/141/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/155/)

**Comment by raquelalvarezbanos** [comment]
Merging, as test failures are unrelated

**Pull request merged**